src/v2/auth.rs

```diff
@@ -215,6 +187,25 @@ fn parse_hdr_bearer(input: &str) -> Result<(String, Option<&str>)> {
     Ok((auth_ep, service))
 }
 
+pub async fn authenticate_client(mut client: Client, login_scope: String) -> Result<Client> {
```

Comment:
Can we move this to a proper method of `&mut Client`?

Reply:
The reason why I wrote this to consume the client is that it's easier in an asynchronous context. I can try again now to change that.

Reply:
Ack, I'm also fine keeping it then.

You can drop the `mut` from the input parameters and re-bind internally. That would make consumers happier.

Reply:
I don't think this `mut` matters to the consumer since the ownership is moved into the function. I did move it to be a method of `Client`, but other than that I left it unchanged.